ugsync/src/main/java/org/apache/ranger/unixusersync/process/PolicyMgrUserGroupBuilder.java

```diff
@@ -812,6 +821,45 @@ private void delXUserGroupInfo(XUserInfo aUserInfo, XGroupInfo aGroupInfo) {
 
 	}
 
+	private MUserInfo updateMUser(String aUserName, List<String> groups) {
+		MUserInfo ret = null;
+		MUserInfo userInfo = new MUserInfo();
+
+		userInfo.setLoginId(aUserName);
+		userInfo.setFirstName(aUserName);
+		userInfo.setLastName(aUserName);
+
+		if (isUserInAdminGroup (groups)) {
+			String[] userRoleList = { "ROLE_SYS_ADMIN" };
+			userInfo.setUserRoleList(userRoleList);
+		}
+
+		if (authenticationType != null && AUTH_KERBEROS.equalsIgnoreCase(authenticationType) && SecureClientLogin.isKerberosCredentialExists(principal, keytab)) {
+			try {
+				Subject sub = SecureClientLogin.loginUserFromKeytab(principal, keytab, nameRules);
+				final MUserInfo result = ret;
+				final MUserInfo userInfoFinal = userInfo;
+				ret = Subject.doAs(sub, new PrivilegedAction<MUserInfo>() {
+					@Override
+					public MUserInfo run() {
+						try {
+							return getUpdatedMUserInfo(userInfoFinal, result);
+						} catch (Exception e) {
+							LOG.error("Failed to update user: ", e);
+						}
+						return null;
+					}
+				});
+				return ret;
+			} catch (Exception e) {
+				LOG.warn("Failed to Authenticate Using given Principal and Keytab: " , e);
```

Comment:
nitpick: spaces...

Reply:
bad usage of copy/paste. Will fix the space issues.